apps/server/src/modules/tool/uc/school-external-tool.uc.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@nestjs/common';
+import { SchoolExternalToolDO } from '@shared/domain/domainobject/external-tool/school-external-tool.do';
+import { Actions, EntityId, Permission, User } from '@shared/domain';
+import { SchoolDO } from '@shared/domain/domainobject/school.do';
+import { AuthorizationService } from '../../authorization';
+import { SchoolExternalToolService } from '../service/school-external-tool.service';
+import { SchoolService } from '../../school';
+
+@Injectable()
+export class SchoolExternalToolUc {
+	constructor(
+		private readonly authorizationService: AuthorizationService,
+		private readonly schoolExternalToolService: SchoolExternalToolService,
+		private readonly schoolService: SchoolService
+	) {}
+
+	async findSchoolExternalTools(
+		userId: EntityId,
+		query: Partial<SchoolExternalToolDO>
```

Comment:
We should not use DOs as parameters for use cases. DOs should exist entirely inside the domain layer. Also the `Partial<T>` helper makes all properties optional. I'm not sure if this is intended.

Reply:
I created a new type which will passed as parameter to the uc and also to the exported services